lib/analyze_config_file_diffs_task.rb

```diff
@@ -23,6 +23,18 @@ def analyze(description)
       "repositories",
       "config_files"
     )
+    online_repo = 0
+    description["repositories"].each do |repo|
+      if !repo["url"].start_with?("dvd:///") &&
```

Comment:
We already have a method in the data model which performs this check. I'd rewrite lines 26-36 as

```ruby
if !description["repositories"].any? { |repo| repo.enabled && !repo.external_medium? }
  raise Machinery::Errors::UpgradeFailed.new("No online repositories available.")
end
```